extensions/machine-learning-services/src/widgets/configTable.ts

```diff
@@ -0,0 +1,182 @@
+		this._loadingComponent.updateProperties({ loading: true });
+		let connection = await this.getCurrentConnection();
+		const speesConfig = await this.addTableRow(constants.mlsExternalExecuteScriptTitle,
+			async () => {
+				return await this._serverConfigManager.isMachineLearningServiceEnabled(connection);
+			}, async (enable) => {
+				await this._serverConfigManager.updateExternalScriptConfig(connection, enable);
+				this.refresh();
+			}
+		);
+		const pythonConfig = await this.addTableRow(constants.mlsPythonLanguageTitle,
+			async () => {
+				return await this._serverConfigManager.isPythonInstalled(connection);
+			}, async () => {
+				await this._serverConfigManager.openInstallDocuments();
+			}
+		);
+		const rConfig = await this.addTableRow(constants.mlsRLanguageTitle,
+			async () => {
+				return await this._serverConfigManager.isRInstalled(connection);
+			}, async () => {
+				await this._serverConfigManager.openInstallDocuments();
+			}
+		);
+		this._statusTable.data = [speesConfig, pythonConfig, rConfig];
+		this._loadingComponent.updateProperties({ loading: false });
+	}
+
+	private async getCurrentConnection(): Promise<azdata.connection.ConnectionProfile> {
+		return await this._apiWrapper.getCurrentConnection();
+	}
+
+	private async addTableRow(configName: string, checkEnabledFunction: () => Promise<boolean>, updateFunction: (enable: boolean) => Promise<void>): Promise<azdata.Component[]> {
+		const isEnabled = await checkEnabledFunction();
+
+		const nameCell = this._modelBuilder.text()
+			.withProperties<azdata.TextComponentProperties>({
+				value: configName,
+				CSSStyles: { 'user-select': 'none', ...constants.cssStyles.text }
+			}).component();
+		const statusIconCell = this._modelBuilder.text()
+			.withProperties<azdata.TextComponentProperties>({
+				value: this.getConfigStatusIcon(isEnabled),
+				ariaRole: 'img',
+				title: this.getConfigStatusTest(isEnabled),
+				CSSStyles: { 'user-select': 'none', ...constants.cssStyles.text }
+			}).component();
+
+		const button = this._modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			label: '',
+			title: ''
+		}).component();
+
+		button.label = this.getLabel(isEnabled);
+		button.onDidClick(async () => {
+			await updateFunction(!isEnabled);
+			const isEnabledNewValue = await checkEnabledFunction();
+			button.label = this.getLabel(isEnabledNewValue);
+		});
+		return [
+			nameCell,
+			statusIconCell,
+			button
+		];
+	}
+
+	private getConfigStatusIcon(enabled: boolean): string {
+		if (enabled) {
+			return '✔️';
+		} else {
+			return '❌';
+		}
+	}
+
+	private getConfigStatusTest(enabled: boolean): string {
+		if (enabled) {
+			return constants.mlsEnableButtonTitle;
+		} else {
+			return constants.mlsDisableButtonTitle;
+		}
+	}
+
+	private getLabel(isEnabled: boolean): string {
+		let enable = constants.mlsEnableButtonTitle;
+		let disable = constants.mlsDisableButtonTitle;
+		let title = enable;
```

Comment:
`return isEnabled ? disable : enable;`

And nit: but you have a lot of these `if (something) return thing1; else return thing2` around which is way more verbose than it needs to be. Just using the ternery statement as in my example is a lot cleaner if you're just returning something immediately.